sdk/core/src/request.rs

```diff
@@ -0,0 +1,66 @@
+use http::{HeaderMap, Method};
+use std::fmt::Debug;
+use url::Url;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Request {
```

Comment:
Why was this moved to a separate module? If we don't want `Request` together with response than we should move `Response` to its own module. I'd prefer there to be an `http` module that `Request` and `Response` were both a part of.

Reply:
In the end request will be much more complicated because it will need to support AsyncRead + AsyncSeek streams as well as Bytes.  That's why I've elected to give it a separated module (but we should also do it for response of course 🤦‍♂️ ).

Reply:
I fine with `Request` being in its own module, but I think then we should have an `http` module which has two submodules: `request` and `response`.